controllers/discovery/discovery_test.go

```diff
@@ -116,7 +116,7 @@ func doTesQueryKubernetesDeploymentWithService(t *testing.T, outputFormat string
 	service.Spec.Type = corev1.ServiceTypeNodePort
 
 	cli := fake.NewClientBuilder().WithRuntimeObjects(deployment, service).Build()
-	ctg := NewServiceCatalog(cli)
+	ctg := NewServiceCatalog(cli, nil, nil)
```

Comment:
Having a `DiscoveryClient` you can get rid of passing `nil` here.

`NewKnativeDiscoveryClient(cli, knativeEven, knativeSvc)` and `NewDiscoveryClient(cli)`. Alternatively, the `NewServiceCatalog` could create this knative clients internally if it only depends on `client`.